> Probably related to the recent LLVM upgrade.
> 
> ```
> unity-test> tests/test_unity_doubles.c:1079:5: error: implicit conversion when passing 'const void *' to parameter of type 'const UNITY_DOUBLE *' (aka 'const double *') is not permitted in C++ [-Werror,-Wimplicit-void-ptr-cast]
> unity-test>  1079 |     TEST_ASSERT_EACH_EQUAL_DOUBLE(1.0, p0, 1);
> unity-test>       |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> ```

oh definitely

unity-test/cavalcade failures unrelated to this PR:
- https://hydra.nixos.org/job/nixpkgs/trunk/unity-test.aarch64-darwin
- https://hydra.nixos.org/build/310769595
- https://cache.nixos.org/log/g09sc9cfx1j36qq7x7izbv5f59c5yynx-unity-test-2.6.1.drv
- https://hydra.nixos.org/job/nixpkgs/trunk/cavalcade.aarch64-darwin

--- a/pkgs/development/python-modules/gst-python/default.nix
+++ b/pkgs/development/python-modules/gst-python/default.nix
@@ -6,6 +6,9 @@
   fetchpatch,
   meson,
   ninja,
+  # TODO: We can get rid of this once `buildPythonPackage` accepts `finalAttrs`.
+  # See: https://github.com/NixOS/nixpkgs/pull/271387
+  gst-python,
 
   pkg-config,
   python,
@@ -42,13 +45,6 @@ buildPythonPackage rec {
     })
   ];
 
-  postPatch = lib.optionalString stdenv.hostPlatform.isDarwin ''
-    # The analytics tests often timeout under load on Darwin (e.g. on Hydra), so remove them
-    substituteInPlace testsuite/meson.build --replace-fail \
-      "['Test analytics', 'test_analytics.py', ['gst-plugins-bad/gst-libs/gst/analytics', 'gst-plugins-base/gst-libs/gst/video']]," \
-      ""
-  '';
-
   # Python 2.x is not supported.
   disabled = !isPy3k;
 
@@ -80,7 +76,19 @@ buildPythonPackage rec {
     "-Dpygi-overrides-dir=${placeholder "out"}/${python.sitePackages}/gi/overrides"
     # Exec format error during configure
     "-Dpython-exe=${python.pythonOnBuildForHost.interpreter}"
+    # This is needed to prevent the project from looking for `gst-rtsp-server`
+    # from `checkInputs`.
+    #
+    # TODO: This should probably be moved at least partially into the Meson hook.
+    #
+    # NB: We need to use `doInstallCheck` here because `buildPythonPackage`
+    # renames `doCheck` to `doInstallCheck`.
+    (lib.mesonEnable "tests" gst-python.doInstallCheck)
   ];
+
+  # Tests are very flaky on Darwin.
+  # See: https://github.com/NixOS/nixpkgs/issues/454955
+  doCheck = !stdenv.hostPlatform.isDarwin;
 
   # `buildPythonPackage` uses `installCheckPhase` and leaves `checkPhase`
   # empty. It renames `doCheck` from its arguments, but not `checkPhase`.

--- a/pkgs/development/python-modules/gst-python/default.nix
+++ b/pkgs/development/python-modules/gst-python/default.nix
@@ -82,9 +82,10 @@ buildPythonPackage rec {
     "-Dpython-exe=${python.pythonOnBuildForHost.interpreter}"
   ];
 
-  # TODO: Meson setup hook does not like buildPythonPackage
-  # https://github.com/NixOS/nixpkgs/issues/47390
-  installCheckPhase = "meson test --print-errorlogs";
+  # `buildPythonPackage` uses `installCheckPhase` and leaves `checkPhase`
+  # empty. It renames `doCheck` from its arguments, but not `checkPhase`.
+  # See: https://github.com/NixOS/nixpkgs/issues/47390
+  installCheckPhase = "mesonCheckPhase";
 
   preCheck = lib.optionalString stdenv.hostPlatform.isDarwin ''
     export DYLD_LIBRARY_PATH="${gst_all_1.gst-plugins-base}/lib"

